widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.kt

```diff
@@ -877,7 +878,7 @@ internal class ChatView(context: Context, attrs: AttributeSet?, defStyleAttr: In
     }
 
     override fun onFileDownloadClick(file: AttachmentFile) {
-        if (ContextCompat.checkSelfPermission(
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S || ContextCompat.checkSelfPermission(
```

Comment:
This function calls the `downloadFileToDownloads` under the hood, which checks for `Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q`, I think it's better to have the same version check in this function.
```java
public Completable downloadFileToDownloads(String fileName, String contentType, InputStream inputStream) {
        if (fileName == null) return Completable.error(new NullPointerException("File name cannot be null"));

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
            return downloadFileToDownloadsAPI29(fileName, contentType, inputStream);
        } else {
            return downloadFileToDownloadsOld(fileName, inputStream);
        }
    }
```

Reply:
https://developer.android.com/training/data-storage#scoped-storage
To give users more control over their files and to limit file clutter, apps that target Android 10 (API level 29) and higher are given scoped access into external storage, or scoped storage, by default. Such apps have access only to the app-specific directory on external storage, as well as specific types of media that the app has created.

Reply:
I'm not sure I'm following. @DavDo Based on your last comment, do you suggest to change the condition to `Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q || ContextCompat.checkSelfPermission`?

Reply:
Yes, permissions are not necessary for versions 29 and above, according to my interpretation of the mentioned functions.

Reply:
Done